src/drivers/intel/ssp/ssp.c

```diff
@@ -870,6 +871,31 @@ static int ssp_trigger(struct dai *dai, int cmd, int direction)
 	return 0;
 }
 
+static int ssp_set_clkctrl(struct dai *dai, struct sof_ipc_dai_clkctrl *clkctrl)
+{
+	int ret = 0;
+
+	dai_info(dai, "ssp_set_clkctrl(): SSP%d clk_en %d clk_id %d",
+		 dai->index, clkctrl->ssp.clk_en, clkctrl->ssp.clk_id);
+
+	/* enable the entire port instead of checking clk_id */
```

Comment:
Yeah, I don't know how to control mclk/bclk separately...

Reply:
I would suggest that we add two helper functions to control mclk and bclk respectively. We count ref count in those helper functions. And we use the helper functions within` ssp_set_clkctrl`, `ssp_port_stop`, and `ssp_port_start`.